numpy/ctypeslib.py

```diff
@@ -401,8 +401,9 @@ def prep_pointer(pointer_obj, shape):
         attach an __array_interface__ property to it if it does not
         yet have one.
         """
-        try: pointer_obj.__array_interface__
+        try: pointer_obj.__array_interface__['shape'] = shape
         except AttributeError: pass
+        except TypeError: return
```

Comment:
The `__array_interface__` variable could be something else. It might be an issue with the user-implementation and thus possible. I just don't want to make an assumption of which is right or wrong. In this method we only prepare `__array_interface__` as good as possible. Just after this `np.array` is called to wrap the pointer and this method will raise a more meaningful error if `__array_interface__` is something that cannot be handled. Additionally `__array_interface__` was not set by `prepare_pointer` in this case, so assuming that setting the shape should work, might be wrong.

If you'd prefer to have the separation of the two possible errors, I can implement it that way. I just don't like it in the sense of readability and verbosity.